cluster_test.go

```diff
@@ -1041,11 +1041,30 @@ func (s *S) TestSocketTimeout(c *C) {
 	// Do something.
 	result := struct{ Ok bool }{}
 	err = session.Run("getLastError", &result)
+	c.Assert(mgo.GetStats().ErrSocketTimeout, Equals, 1)
 	c.Assert(err, ErrorMatches, ".*: i/o timeout")
 	c.Assert(started.Before(time.Now().Add(-timeout)), Equals, true)
 	c.Assert(started.After(time.Now().Add(-timeout*2)), Equals, true)
 }
 
+func (s *S) TestSocketStatsReadEOF(c *C) {
+	if *fast {
+		c.Skip("-fast")
+	}
+
+	session, err := mgo.Dial("localhost:40001")
+	c.Assert(err, IsNil)
+	defer session.Close()
+
+	s.Stop("localhost:40001")
```

Comment:
Just a thought: is Stop() synchonously stopping it such that the socket is definitely closed when it returns, or is there a chance of this being flaky?

Reply:
stop kills the process running on localhost:40001 addr synchronously - and then line right after that s.StartAll() would start the process again